components/service/glean/src/test/java/mozilla/components/service/glean/scheduler/MetricsPingTest.kt

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.scheduler
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import java.util.concurrent.TimeUnit
+
+@RunWith(RobolectricTestRunner::class)
+class MetricsPingSchedulerTest {
+    // NOTE: Using a "real" ApplicationContext here so that it will have
+    // a working SharedPreferences implementation
+    private val metricsPingScheduler = MetricsPingScheduler(RuntimeEnvironment.application.applicationContext)
+
+    @Before
+    fun setup() {
+        metricsPingScheduler.clearSchedulerData()
+    }
+
+    @Test
+    fun `metrics ping canSendPing() returns true if no timestamp was recorded in shared prefs`() {
+        assertTrue(metricsPingScheduler.canSendPing())
+    }
+
+    @Test
+    fun `MetricsPingScheduler updatePingTimestamp() correctly stores values`() {
+        // First, try storing the default which is the current system time
+        val ts = metricsPingScheduler.updateSentTimestamp()
```

Comment:
Here we're testing the default value, which I think can be left out of the test coverage :) This way we could drop the return value from the function.